src/utils/HashlinkEncoder.ts

```diff
@@ -68,35 +67,49 @@ export class HashlinkEncoder {
   }
 
   /**
-   *
    * Validates a hashlink
    *
    * @param hashlink the hashlink that needs validating
    *
    * @returns a boolean whether the hashlink is valid
    *
    * */
-
   public static isValid(hashlink: string): boolean {
     const hashlinkList = hashlink.split(':')
-    const validMultibase = MultibaseEncoder.isValid(hashlinkList[1])
-    if (!validMultibase) {
+    const validMultiBase = MultiBaseEncoder.isValid(hashlinkList[1])
+    if (!validMultiBase) {
       return false
     }
-    const { data } = MultibaseEncoder.decode(hashlinkList[1])
-    const validMultihash = MultihashEncoder.isValid(data)
-    return validMultibase && validMultihash ? true : false
+    const { data } = MultiBaseEncoder.decode(hashlinkList[1])
+    const validMultiHash = MultiHashEncoder.isValid(data)
+    return validMultiBase && validMultiHash ? true : false
+  }
+
+  public static encodeAttachment(
+    attachment: Attachment,
+    hashAlgorithm: 'sha2-256' = 'sha2-256',
+    baseName: BaseName = 'base58btc'
+  ) {
+    if (attachment.data.sha256) {
+      return `hl:${attachment.data.sha256}`
+    } else if (attachment.data.base64) {
+      return this.encode(BufferEncoder.fromBase64(attachment.data.base64), {}, hashAlgorithm, baseName)
+    } else if (attachment.data.json) {
+      return this.encode(JsonEncoder.toBuffer(attachment.data.json), {}, hashAlgorithm, baseName)
+    } else {
+      throw new AriesFrameworkError(`Attachment: (${attachment.id}) has no data to create a link with`)
+    }
```

Comment:
This doesn't seem like the right place. Would rather see this in an attachment util or something